### PR TITLE
Change color acording to the level

### PR DESCRIPTION
**Pull request by shargon**
image

### CODE REVIEW

**Comment by shargon** [comment]
We should define the colors, feel free to change it

**Comment by vncoelho** [comment]
Looks good to me, @shargon. You are an artist.

File: neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -175,8 +175,21 @@ private void InitializeConsensus(byte viewNumber)
...
+                case LogLevel.Error: ConsoleColorSet.Error.Apply(); break;
+                case LogLevel.Fatal: ConsoleColorSet.Fatal.Apply(); break;
+                case LogLevel.Info: ConsoleColorSet.Info.Apply(); break;
+                case LogLevel.Warning: ConsoleColorSet.Warning.Apply(); break;
+            }
+
             Console.WriteLine($"[{DateTime.Now.TimeOfDay:hh\\:mm\\:ss\\.fff}] {message}");
             Plugin.Log(nameof(ConsensusService), level, message);
+
+            currentColor.Apply();
```
Comment (vncoelho) [inline neo/Consensus/ConsensusService.cs]: @shargon, this way all the time we are going to set the log? In every message?

Comment (shargon) [reply neo/Consensus/ConsensusService.cs]: We need to back to the previous color

Comment (vncoelho) [reply neo/Consensus/ConsensusService.cs]: Onh man...But there is an other solution @igormcoelho

Comment (igormcoelho) [reply neo/Consensus/ConsensusService.cs]: This is good Vitor... it's just stream set/unset, it's bullet fast.

File: neo/Consensus/ConsensusService.cs [hunk 2]
```diff
@@ -175,8 +175,21 @@ private void InitializeConsensus(byte viewNumber)
 
         private void Log(string message, LogLevel level = LogLevel.Info)
         {
+            var currentColor = new ConsoleColorSet();
```
Comment (igormcoelho) [inline neo/Consensus/ConsensusService.cs]: Only thing I don't like here is this Dynamic memory criation for object... couldn't this all be static? it's such a simple thing...

Comment (shargon) [reply neo/Consensus/ConsensusService.cs]: If you want, we could store the initial color in a static var

**Comment by shargon** [comment]
@igormcoelho When `Log` is called, you can be writing a password (Yellow), or it could be listing your addresses (White), both colors are different, so what is your proposal for the static var?

**Comment by igormcoelho** [comment]
I mean, instead of creating a `new ColorObject` for every log, you could just invoke static coloring methods, no?

**Comment by erikzhang** [comment]
Can we move this to a plugin?

**Comment by shargon** [comment]
I think that the logic affected is here, in `ConsensusService.Log`

**Comment by erikzhang** [comment]
I mean, move all the log related stuff to the plugin.

**Comment by shargon** [comment]
Take a look at https://github.com/neo-project/neo-plugins/pull/112, i will change this for remove the Console.Write

**Comment by shargon** [comment]
Problems with CRLF

**Pull request closed without merging**